### ISSUE TITLE
update changelog management

### ISSUE THREAD

**Issue by trife** [PhenoApps/Field-Book#605]
- [x] create [changelog.md](https://keepachangelog.com/) that contains all merges
- [x] extract information from pull requests to update changelog.md
- [x] convert entries to changelog.xml entries for release

**Comment by bellerbrock** [comment]
The linked tool seems to follow a different paradigm than our existing workflows. It assumes tags trigger releases and that there are many PRs in between each release.

So to start at least I've tried some simpler updates to the github release workflow in the [automation testing repo](https://github.com/FieldPheno/Field-Book-Automation/blob/0f78e0caa68a17fca0e03506e8cc24d7aa1267f6/.github/workflows/github-release.yml#L61) by using the git api to:

- extract the changelog line from PR template on PR merge commit
then
- use the line plus PR # as the new github release body (if minor or major) or append it to the previous release body (if patch)
- add the line in a new section of changelog.xml (if minor or major) or append to the latest section (if patch)

Retrieving the pr body is working fine, but extracting the relevant line is not working. An easier method would be using the commit message/PR title as the changelog line (which is the current default). Would that be fine?

**Comment by trife** [comment]
`RELEASE_NOTE=$(echo "${{ github.event.pull_request.body }}" | awk '/```release-note/{flag=1;next}/```/{flag=0}flag')`

**Comment by bellerbrock** [comment]
Not having any luck with that either, but might be missing something. Discuss at 2?

**Comment by trife** [comment]
Yeah. It's breaking for a git-related reason, though, so it's parsing correctly afaict

**Comment by trife** [comment]
Two new workflows to try:
A:
On PR merged workflow
1. [on-pr-merged](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges) extract `release-notes` from body: `RELEASE_NOTE=$(echo "${{ github.event.pull_request.body }}" | awk '/```release-note/{flag=1;next}/```/{flag=0}flag')`
2. write to changelog.xml
3. on-gh-release do a diff on the changelog.xml and add the extracted string to the gh release

B:
1. PR created
2. [on-review-approved](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved) `release-notes` tag extracted and added to changelog.xml along with tag determined by type of change selected
4. on-gh-release do a diff on the changelog.xml and add the extracted string to the gh release

**Comment by trife** [comment]
here's an idea- not sure how good it is.

1. on-pr check for updates to changelog.xml.
2. if changelog.xml hasn't been updated, extract release-notes text and write to changelog.xml in current release block.
3. on-pr-merge
